test/runtests.jl

```diff
@@ -1,8 +1,20 @@
 using Diffusion
 using Test
+using StaticArrays
 
 @testset "Diffusion.jl" begin
     # Write your tests here.
-    @test absorb_boundry(1,2) == 3
+    @test absorb_boundry(SA[3,4],6) == SA[3,4]
+    @test absorb_boundry(SA[3,4],5) === nothing
+    @test absorb_boundry(SA[3,4],4) === nothing 
+    @test slope(SA[1,2],SA[3,4]) == 1
+    @test midpoint(SA[1,1],SA[3,3]) == SA[2,2]
+    @test isapprox(circleintersect(SA[1,0], SA[7,0], 2),SA[2,0])
+    @test rslope(SA[2,3]) == 3/2
+    @test perpslope(3) == -1/3
+    @test solve(1,-2,0,3) == SA[1,1]
+    @test posFinal(SA[1,2],SA[1.5,1.5]) == SA[2,1]
+    @test isapprox(reflect_boundary(SA[0,2], SA[4,2], sqrt(8)), SA[2,0])
```

Comment:
I might consider trying several values here. Are you familiar with [polar coordinates](https://en.wikipedia.org/wiki/Polar_coordinate_system)? Reflection from a circular boundary is quite easy if points are expressed in polar coordinates, because you can just reflect the radius and leave the angle constant. That might allow you to write some more thorough tests, like
```
for r in 0.5:0.1:1.2, θ in 0:π/8:2π
    pt = SA[r*cos(θ), r*sin(θ)]
    rbound = 0.8
    rreflect = 2*rbound - r
    @test reflect_boundary(pt, rbound) ≈ r <= rbound ? pt : SA[rreflect*cos(θ), rreflect*sin(θ)]
end
```
This loops over several different angles and radii and checks the behavior for all of them. Note it checks the version I had in mind that doesn't require the initial point, but you could perhaps modify it. If nothing else, it shows another testing style that checks functions a bit more thoroughly. A few targeted values are fine, though.

Reply:
I went ahead and only checked a few spots-whenever I tried to run the more complicated version of the test, every single one errored.